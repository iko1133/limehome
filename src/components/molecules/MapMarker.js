import React from "react";
import { Marker } from "react-native-maps";
import styled from "styled-components/native";

const Container = styled(Marker)`
  align-items: center;
`;

const MarkerContainer = styled.View`
  background-color: ${({ theme, isSelected }) =>
    isSelected
      ? theme.SELECTED_MARKER_BACKGROUND_COLOR
      : theme.MARKER_BACKGROUND_COLOR};
  border-radius: ${({ isSelected }) => (isSelected ? 10 : 6)}px;
  padding: 12px 8px;
`;

const MarkerText = styled.Text`
  color: ${({ theme }) => theme.MARKER_TEXT_COLOR};
  font-size: 14px;
  font-weight: 400;
`;

const ArrowDown = styled.View`
  background-color: ${({ theme, isSelected }) =>
    isSelected
      ? theme.SELECTED_MARKER_BACKGROUND_COLOR
      : theme.MARKER_BACKGROUND_COLOR};
  height: 10px;
  margin-top: -5px;
  transform: rotate(45deg);
  width: 10px;
`;

const MapMarker = ({ property, onPress, isSelected, ...otherProps }) => {
  return (
    <Container coordinate={property.location} onPress={onPress} {...otherProps}>
      <MarkerContainer isSelected={isSelected}>
        <MarkerText>{10}â‚¬</MarkerText>
      </MarkerContainer>
      <ArrowDown isSelected={isSelected} />
    </Container>
  );
};

export default MapMarker;
