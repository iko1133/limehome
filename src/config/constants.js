export const MAP_DEFAULT_COORDINATES = {
  latitude: 52.520008,
  longitude: 13.404954,
  latitudeDelta: 0.15,
  longitudeDelta: 0.15,
};
