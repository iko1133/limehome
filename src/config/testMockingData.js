export const PROPERTIES = [
  {
    id: 99,
    name: "berlin schlüterstraße",
    city: "berlin",
    city_id: 32,
    street: "schlüterstraße",
    location: {
      lat: 52.50176279999999,
      lng: 13.3164893,
      city: "Berlin",
    },
    distance: 6.321157219990815,
    images: [
      {
        url: "https://limehome.imgix.net/properties/99/9fee5516-9d41-41f7-95ea-6ac72c49fb86.jpg",
        is_portrait: false,
        position: 0,
        unit_group_ids: [16],
        tags: [],
      },
    ],
    lowest_price_per_night: null,
    lowest_price_per_month: null,
  },
  {
    id: 152,
    name: "berlin werftstraße",
    city: "berlin",
    city_id: 32,
    street: "werftstraße ",
    location: {
      lat: 52.52259309999999,
      lng: 13.3589213,
      city: "Berlin",
    },
    distance: 3.127749150912429,
    images: [
      {
        url: "https://limehome.imgix.net/properties/152/bff791de-57fa-450b-869d-0a7bfac7a55c.jpg",
        is_portrait: false,
        position: 0,
        unit_group_ids: [178],
        tags: [],
      },
    ],
    lowest_price_per_night: null,
    lowest_price_per_month: null,
  },
];

export const FIXED_PROPERTIES = [
  {
    id: 99,
    name: "berlin schlüterstraße",
    city: "berlin",
    city_id: 32,
    street: "schlüterstraße",
    location: {
      latitude: 52.50176279999999,
      longitude: 13.3164893,
      city: "Berlin",
    },
    distance: 6.321157219990815,
    images: [
      {
        url: "https://limehome.imgix.net/properties/99/9fee5516-9d41-41f7-95ea-6ac72c49fb86.jpg",
        is_portrait: false,
        position: 0,
        unit_group_ids: [16],
        tags: [],
      },
    ],
    lowest_price_per_night: null,
    lowest_price_per_month: null,
  },
  {
    id: 152,
    name: "berlin werftstraße",
    city: "berlin",
    city_id: 32,
    street: "werftstraße ",
    location: {
      latitude: 52.52259309999999,
      longitude: 13.3589213,
      city: "Berlin",
    },
    distance: 3.127749150912429,
    images: [
      {
        url: "https://limehome.imgix.net/properties/152/bff791de-57fa-450b-869d-0a7bfac7a55c.jpg",
        is_portrait: false,
        position: 0,
        unit_group_ids: [178],
        tags: [],
      },
    ],
    lowest_price_per_night: null,
    lowest_price_per_month: null,
  },
];
