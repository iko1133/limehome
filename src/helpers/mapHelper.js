// Copies given object, except, switches names of the location property's properties for the
// marker objects to function properly
// lgn to longitude
// lat to latitude
export const fixObjectLocationProperty = (object) => ({
  ...object,
  location: {
    ...object.location,
    longitude: object.location?.lng,
    latitude: object.location?.lat,
  },
});
