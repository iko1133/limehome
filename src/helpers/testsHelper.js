import { ThemeProvider } from "styled-components/native";

import { lightTheme } from "styles/theme";

/**
 * Wraps given children into a ThemeProvider with lightTheme set as the main theme
 * @param {object} children - children to be rendered inside ThemeProvider
 */
export const WrapInThemeProvider = (children) => (
  <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
);
