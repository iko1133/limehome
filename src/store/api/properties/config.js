// list of urls needed
const config = {
  propertiesList: "properties/v1/public/properties/",
  propertyDetails: "properties/v1/public/properties/",
};

export default config;
