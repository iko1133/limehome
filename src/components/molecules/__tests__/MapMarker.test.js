import React from "react";
import renderer from "react-test-renderer";

import { WrapInThemeProvider } from "helpers/testsHelper";
import { FIXED_PROPERTIES } from "config/testMockingData";
import MapMarker from "../MapMarker";

describe("Testing MapMarker", () => {
  it("Renders map marker with property 0 on it", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(<MapMarker property={FIXED_PROPERTIES[0]} />)
      )
    ).toMatchSnapshot());

  it("Renders map marker with property 1 on it, but is also selected", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(
          <MapMarker property={FIXED_PROPERTIES[1]} isSelected />
        )
      )
    ).toMatchSnapshot());
});
