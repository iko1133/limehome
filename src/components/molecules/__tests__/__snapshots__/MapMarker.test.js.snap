// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MapMarker Renders map marker with property 0 on it 1`] = `
<AIRMapMarker
  coordinate={
    Object {
      "city": "Berlin",
      "latitude": 52.50176279999999,
      "longitude": 13.3164893,
    }
  }
  onPress={[Function]}
  stopPropagation={false}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "position": "absolute",
      },
      Array [
        Object {
          "alignItems": "center",
        },
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#4B4B4D",
          "borderRadius": 6,
          "paddingBottom": 12,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 12,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#F7F1E9",
            "fontSize": 14,
            "fontWeight": "400",
          },
        ]
      }
    >
      10
      €
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#4B4B4D",
          "height": 10,
          "marginTop": -5,
          "transform": Array [
            Object {
              "rotate": "45deg",
            },
          ],
          "width": 10,
        },
      ]
    }
  />
</AIRMapMarker>
`;

exports[`Testing MapMarker Renders map marker with property 1 on it, but is also selected 1`] = `
<AIRMapMarker
  coordinate={
    Object {
      "city": "Berlin",
      "latitude": 52.52259309999999,
      "longitude": 13.3589213,
    }
  }
  onPress={[Function]}
  stopPropagation={false}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "position": "absolute",
      },
      Array [
        Object {
          "alignItems": "center",
        },
      ],
    ]
  }
  tracksViewChanges={true}
>
  <View
    isSelected={true}
    style={
      Array [
        Object {
          "backgroundColor": "#B26422",
          "borderRadius": 10,
          "paddingBottom": 12,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 12,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#F7F1E9",
            "fontSize": 14,
            "fontWeight": "400",
          },
        ]
      }
    >
      10
      €
    </Text>
  </View>
  <View
    isSelected={true}
    style={
      Array [
        Object {
          "backgroundColor": "#B26422",
          "height": 10,
          "marginTop": -5,
          "transform": Array [
            Object {
              "rotate": "45deg",
            },
          ],
          "width": 10,
        },
      ]
    }
  />
</AIRMapMarker>
`;
